src/dlsnode/util/aio/internal/JobQueue.d

```diff
@@ -129,6 +115,68 @@ public static struct Job
     ****************************************************************/
 
     private bool is_slot_free;
+
+    /****************************************************************
+
+        Buffer for the thread to fill and to copy to the user buffer
+        from.
+
+    ****************************************************************/
+
+    public void[] recv_buffer;
+
+    /***************************************************************
+
+        User buffer to copy the results to. NOTE: the reason to have
+        recv_buffer and user_buffer separate is because it might happen
+        that user discards the job and the buffer, while thread is
+        writting into it.
+
+    ****************************************************************/
+
+    public void[] user_buffer;
```

Comment:
I wonder if there's a nicer way to handle job descriptions. Not all fields are used by all job types, which makes this a bit confusing.

A union would be one way. (Another would be to serialize job structs, like swarm neo does for requests, but that'd probably be overkill here.)

Reply:
(I wouldn't suggest changing this now, but something to think about for a future PR.)

Reply:
> I wonder if there's a nicer way to handle job descriptions. Not all fields are used by all job types, which makes this a bit confusing.

This is what I also thought it should be worked on. I'll address this in the PR following this one.